tobago-example/tobago-example-demo/src/test/java/org/apache/myfaces/tobago/example/demo/integration/FrontendBase.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.myfaces.tobago.example.demo.integration;
+
+import org.apache.commons.lang3.time.DurationFormatUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+abstract class FrontendBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  @SuppressWarnings("rawtypes") // this is how to use testcontainers
+  @Container
+  private static final GenericContainer SELENIUM_FIREFOX =
+      new GenericContainer<>(DockerImageName.parse("henningn/selenium-standalone-firefox"))
+          .withExposedPorts(4444);
+
+  @SuppressWarnings("rawtypes") // this is how to use testcontainers
+  @Container
+  private static final GenericContainer TOMCAT =
+      new GenericContainer(DockerImageName.parse("myfaces/tobago-example-demo"))
+          .withExposedPorts(8080);
```

Comment:
We need an own network otherwise you can't access the tomcat from the selenium grid and we should be able to log the output from the container

```suggestion
  private static final Network NETWORK = Network.newNetwork();
  private static final String TOMCAT_ALIAS = "tomcat";

  public static final int SELENIUM_PORT = 4444;
  public static final int TOMCAT_PORT = 8080;

  @Container
  private static final GenericContainer<?> SELENIUM_FIREFOX =
      new GenericContainer<>(DockerImageName.parse("henningn/selenium-standalone-firefox"))
          .withNetwork(NETWORK).withNetworkAliases("selenium").withExposedPorts(SELENIUM_PORT)
          //.withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("selenium")).withSeparateOutputStreams())
          ;

  @Container
  private static final GenericContainer<?> TOMCAT =
      new GenericContainer<>(DockerImageName.parse("myfaces/tobago-example-demo"))
          .withNetwork(NETWORK).withNetworkAliases(TOMCAT_ALIAS).withExposedPorts(TOMCAT_PORT)
          .waitingFor(Wait.forHttp("/").forPort(TOMCAT_PORT))
          //.withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("tomcat")).withSeparateOutputStreams())
          ;
```